submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/FileListResource.kt

```diff
@@ -14,11 +16,17 @@ import org.springframework.web.bind.annotation.RestController
 @RequestMapping("/submissions/fileLists")
 class FileListResource(
     private val sourceGenerator: SourceGenerator,
-    private val fileListValidator: FileListValidator
+    private val fileListValidator: FileListValidator,
+    private val onBehalfUtils: OnBehalfUtils
 ) {
     @PostMapping("/validate")
     fun validateFileList(
         @BioUser user: SecurityUser,
-        @RequestParam(FILE_LIST_NAME) fileListName: String
-    ) = fileListValidator.validateFileList(fileListName, sourceGenerator.userSources(user))
+        onBehalfRequest: OnBehalfRequest?,
+        @RequestParam(FILE_LIST_NAME) fileListName: String,
+    ) {
+        val onBehalfUser = onBehalfRequest?.let { onBehalfUtils.getOnBehalfUser(it) }
+
+        fileListValidator.validateFileList(fileListName, sourceGenerator.userSources(user, null, onBehalfUser))
```

Comment:
So here, instead of leaving the onBehalf logic to the source generator, you should do something like

```
val onBehalfUser = onBehalfRequest?.let { onBehalfUtils.getOnBehalfUser(it) }
val userSources = sourceGenerator.userSources(user)
val onBehalfSources = sourceGenerator.userSources(onBehalfUser)

fileListValidator.validateFileList(fileListName, ComposedFilesSource(userSources, onBehalfUserSources))
```